default.nix

```diff
@@ -60,20 +60,19 @@
 }:
 
 let
-  hnix-store-src = pkgs.fetchFromGitHub {
-    owner = "haskell-nix";
-    repo = "hnix-store";
-    rev = "0.1.0.0";
-    sha256 = "1z48msfkiys432rkd00fgimjgspp98dci11kgg3v8ddf4mk1s8g0";
-  };
-
-  overlay = pkgs.lib.foldr pkgs.lib.composeExtensions (_: _: {}) [
-    (import "${hnix-store-src}/overlay.nix")
-    (self: super: with pkgs.haskell.lib; {
 
-      semialign         = super.semialign_1_1;
+  #  2020-05-23: NOTE: Currently HNix-store needs no overlay
+  # hnix-store-src = pkgs.fetchFromGitHub {
```

Comment:
Nixpkgs rolls differently to Hackage rolls.

Nixpkgs branches tend to have the build descriptions that work only for a set of versions that build the same as the latest version, or those old important versions should receive a different name for a package description.

We do not know how things going inside `hnix-store`, until now I still yet to read at least some of their source code.

But we know that if some issues happen between Hackage and Nixpkgs for them - they would put it in hierarchy that we can load from that `overlay.nix` package, we effectively would build a chain `hackage -> nixpkgs -> hnix-store overlay -> hnix-store build` to be able to build it "their current way".

Because Nix language is hard to remember, and we may need that line in the future from time to time - better to live it as a comment, or even compose it into a switch in the header of the file.

Reply:
(I clicked "unresolved" just to to ensure that you see my response)

Thanks for the explanation! Generally, I'm not a huge fan of commented-out code. If we need this again, we can also simply look at the Git history. But you have more experience with Nix so I defer to your judgement. :)